sycl/include/sycl/reduction.hpp

```diff
@@ -1515,6 +1536,8 @@ template <> struct NDRangeReduction<reduction::strategy::range_basic> {
     using Name = __sycl_reduction_kernel<reduction::MainKrn, KernelName,
                                          reduction::strategy::range_basic>;
 
+    WGSize = std::min(WGSize, reduGetPreferredKernelWGSize<Name>(Queue));
+
     CGH.parallel_for<Name>(NDRange, Properties, [=](nd_item<1> NDId) {
```

Comment:
If we are recalculating `WGSize` now based on the named kernel's info query, we likely need to update the `NDRange` for the kernel dispatch here and in all of the updated reduction strategy implementations specialisations.